src/Cms/FileActions.php

```diff
@@ -57,16 +57,19 @@ public function changeName(string $name, bool $sanitize = true): static
 			// rename the main file
 			F::move($oldFile->root(), $newFile->root());
 
-			if ($newFile->kirby()->multilang() === true) {
-				foreach ($newFile->translations() as $translation) {
-					$translationCode = $translation->code();
-
-					// rename the content file
-					F::move($oldFile->contentFile($translationCode), $newFile->contentFile($translationCode));
-				}
-			} else {
+			$translations =
+				$newFile->kirby()->multilang() ?
+				$newFile->translations()->values(
+					fn ($translation) => $translation->code()
+				) :
+				['default'];
```

Comment:
Can we make this a method in the `Languages` collection class to make the logic DRY?

Reply:
Just that some to `$kirby->languages()` other parts do `$model->translations()`

Reply:
We just need the codes and those are always the same no matter if we access them via languages or translations:

https://github.com/getkirby/kirby/blob/6b5dda600472b59a3a779f7e6b8e9e130cc414a1/src/Cms/ModelWithContent.php#L602